src/standardMethods.ts

```diff
@@ -127,6 +127,29 @@ export function generateListNoCursorEndpoint<ResponseType>(
   };
 }
 
+/** @hidden */
+export function generateListNoCursorEndpointWithQueryParams<
+  RequestParams,
+  ResponseType
+>(
+  axiosInstance: AxiosInstance,
+  resourcePath: string,
+  metadataMap: MetadataMap
+) {
+  return async function list(params?: RequestParams): Promise<ResponseType[]> {
+    const response = await rawRequest<ItemsResponse<ResponseType>>(
+      axiosInstance,
+      {
+        method: 'get',
+        url: resourcePath,
+        params,
+      },
+      true
+    );
+    return metadataMap.addAndReturn(response.data.items, response);
+  };
+}
+
```

Comment:
you have the same method but with no params
why not making params optional?
problems dynamic typing?

Reply:
Yes, dynamic typing.. 🤔 Haven't found a good way to do it